Check/Codebase/PhpCodeSniffer.php

```diff
@@ -0,0 +1,171 @@
+        if ($rows++ > 0) {
+          $ret_val .= PHP_EOL;
+          if (!drush_get_option('json')) {
+            $ret_val .= str_repeat(' ', 4);
+          }
+        }
+        $ret_val .= dt('Filename: @filename, Violations: @total', array(
+          '@filename' => $filename,
+          '@total' => count($violations),
+        ));
+        foreach ($violations as $violation) {
+          $ret_val .= PHP_EOL;
+          if (!drush_get_option('json')) {
+            $ret_val .= str_repeat(' ', 6);
+          }
+          $line = $violation['line'];
+          $column = $violation['column'];
+          $severity = $violation['severity'];
+          $message = $violation['message'];
+          $ret_val .= "$line, $column : $severity - $message";
+        }
+      }
+    }
+    return $ret_val;
+
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getAction().
+   */
+  public function getAction() {}
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\calculateScore().
+   */
+  public function calculateScore() {
+    // Get the path of phpcs.
+    $phpcs_path = $this->getExecPath('phpcs');
+    if ($phpcs_path === SiteAuditCheckAbstract::AUDIT_CHECK_SCORE_INFO) {
+      $this->registry['phpcs_path'] = $phpcs_path;
+      return $phpcs_path;
+    }
+    // Get the custom code paths.
+    $custom_code = $this->getCustomCodePaths();
+    if ($custom_code === SiteAuditCheckAbstract::AUDIT_CHECK_SCORE_INFO) {
+      $this->registry['custom_code'] = $custom_code;
+      return $custom_code;
+    }
+    // Get options.
+    // TODO: put in exhaustive list of ignore patterns.
+    $valid_options = array(
+      'extensions' => 'php,module,inc,install,test,profile,theme,css,info,txt',
+      'ignore' => '*/modules/features/',
+      'standard' => SITE_AUDIT_BASE_PATH . '/vendor/drupal/coder/coder_sniffer/Drupal',
+    );
+    $options = $this->getOptions($valid_options, 'phpcs_');
+    // Check if 'standard' is a valid directory.
+    if (!is_dir($options['standard'])) {
+      $this->registry['phpcs_standard'] = $options['standard'];
+      return SiteAuditCheckAbstract::AUDIT_CHECK_SCORE_INFO;
+    }
+    $temp_file = tempnam(sys_get_temp_dir(), 'site_audit');
+    $option_string = " --report=checkstyle";
+    foreach ($options as $option => $value) {
+      $option_string .= " --$option";
+      if ($value !== TRUE) {
+        $option_string .= "=$value";
+      }
+    }
+    // Suppress XML errors which will be handled by try catch instead.
+    libxml_use_internal_errors(TRUE);
+
+    foreach ($custom_code as $path) {
+      $command = $phpcs_path . ' ' . $path . $option_string;
+      $process = new Process($command);
+      $process->run();
+      try {
+        $output = new SimpleXMLElement($process->getOutput());
+        foreach ($output as $file) {
+          foreach ($file as $violation) {
+            $this->registry['phpcs_out'][(String) $file[0]['name']][] = $violation;
+          }
+        }
+      }
+      catch (Exception $e) {
```

Comment:
Log an error using `drush_set_message` level `debug` with the exception text, if it can't read the XML file then it should log it.